src/pcre2_compile.c

```diff
@@ -10568,6 +10587,32 @@ if ((options & PCRE2_LITERAL) == 0)
             else limit_depth = c;
           skipatstart = ++pp;
           break;
+
+          case PSO_OPTMZ:
+          optim_flags &= ~(p->value);
+
+          /* For backward compatibility the three original VERBs to disable
+          optimizations need to also update the corresponding external option. */
+
+          switch(p->value)
+            {
+            case PCRE2_OPTIM_AUTO_POSSESS:
+            cb.external_options |= PCRE2_NO_AUTO_POSSESS;
```

Comment:
Is it ok to override cb? I don't know its purpose.

Reply:
Not sure what you mean by overwriting `cb`, only one bit on the extenal_options is being turned on, just like it was done before (for backward compatibility).

The code doesn't use those bits anymore (neither the ones set in the compile options) but instead relies on the new "optimization flags". Once pcre2test support is added for the new API and start being used that might become more obvious.